apps/meteor/tests/unit/server/services/omnichannel/queue.tests.ts

```diff
@@ -0,0 +1,462 @@
+			expect(queue.reconciliation.calledOnce).to.be.true;
+		});
+		it('should call removeInquiry when findOneById returns a room that was closed', async () => {
+			models.LivechatRooms.findOneById.returns({ _id: 'rid', closedAt: new Date() });
+
+			const queue = new OmnichannelQueue();
+			queue.reconciliation = Sinon.stub().returns(true);
+			expect(await queue.processWaitingQueue('department1', mockedInquiry)).to.be.true;
+			expect(queue.reconciliation.calledOnce).to.be.true;
+		});
+		it('should call delegateInquiry when prechecks are met and return false if inquiry was not served', async () => {
+			models.LivechatRooms.findOneById.returns({ _id: 'rid' });
+			delegateInquiry.returns({});
+
+			const queue = new OmnichannelQueue();
+			expect(await queue.processWaitingQueue('department1', mockedInquiry)).to.be.false;
+			expect(delegateInquiry.calledOnce).to.be.true;
+		});
+		it('should call delegateInquiry and return true if inquiry was served', async () => {
+			models.LivechatRooms.findOneById.returns({ _id: 'rid' });
+			delegateInquiry.returns({ _id: 'rid', servedBy: { _id: 'agentId' } });
+
+			const queue = new OmnichannelQueue();
+			expect(await queue.processWaitingQueue('department1', mockedInquiry)).to.be.true;
+			expect(delegateInquiry.calledOnce).to.be.true;
+		});
+		it('should call dispatchAgentDelegated if inquiry was served (after 1s)', async () => {
+			models.LivechatRooms.findOneById.returns({ _id: 'rid' });
+			delegateInquiry.returns({ _id: 'rid', servedBy: { _id: 'agentId' } });
+
+			const queue = new OmnichannelQueue();
+			expect(await queue.processWaitingQueue('department1', mockedInquiry)).to.be.true;
+			expect(delegateInquiry.calledOnce).to.be.true;
+			clock.tick(1000);
+			expect(dispatchAgentDelegated.calledOnce).to.be.true;
+		});
+	});
+	describe('execute', () => {
+		beforeEach(() => {
+			license.shouldPreventAction.reset();
+			queueLogger.debug.reset();
+		});
+
+		after(() => {
+			license.shouldPreventAction.reset();
+			queueLogger.debug.reset();
+		});
+
+		it('should return if service is not running', async () => {
+			const queue = new OmnichannelQueue();
+			queue.running = false;
+			expect(await queue.execute()).to.be.undefined;
+		});
+		it('should return undefined if license is over mac limits', async () => {
+			license.shouldPreventAction.returns(true);
+
+			const queue = new OmnichannelQueue();
+			queue.running = true;
+			expect(await queue.execute()).to.be.undefined;
+			expect(license.shouldPreventAction.calledOnce).to.be.true;
+		});
+		it('should try to process a queue if license is not over mac limits', async () => {
+			license.shouldPreventAction.returns(false);
+
+			const queue = new OmnichannelQueue();
+			queue.running = true;
+			queue.nextQueue = Sinon.stub();
+			await queue.execute();
+
+			expect(queue.nextQueue.calledOnce).to.be.true;
+			expect(queueLogger.debug.calledOnce).to.be.true;
```

Comment:
I think just the assertion with `calledWith` is good enough, isn't it?

Reply:
Yeah, don't think that's something to worry tho.